src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1245,22 +1530,29 @@ public override BoundNode VisitLocalDeclaration(BoundLocalDeclaration node)
                 if (valueType.HasNullType)
                 {
                     Debug.Assert(type.IsErrorType());
-                    valueType = type.ToTypeWithState();
+
+                    if (!(initializer.Type is null))
```

Comment:
>if (!(initializer.Type is null)) [](start = 20, length = 32)

Why add this condition?

Reply:
I'll add a comment, but the gist of it is that the initializer has a type and `valueType` does not, then we would be changing the type of `initializer` from `ErrorType` to `null`.

---
In reply to: [265699529](https://github.com/dotnet/roslyn/pull/34017#discussion_r265699529) [](ancestors = 265699529)

Reply:
That sounds like a bug in the particular `Visit` method for the initializer. Please consider logging a bug and adding a comment here.

---
In reply to: [265763409](https://github.com/dotnet/roslyn/pull/34017#discussion_r265763409) [](ancestors = 265763409,265699529)